java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -0,0 +1,189 @@
+      if (programPosition.block.successors().isEmpty()) {
+        endOfExecutionPath.add(node);
+        continue;
+      }
+
+      if (programPosition.i < programPosition.block.elements().size()) {
+        // process block element
+        executeInstruction((BytecodeCFGBuilder.Instruction) programPosition.block.elements().get(programPosition.i));
+      } else {
+        // process block exit, which is unconditional jump such as goto-statement or return-statement
+        handleBlockExit(programPosition);
+      }
+    }
+
+    handleEndOfExecutionPath();
+    executeCheckEndOfExecution();
+    // Cleanup:
+    workList = null;
+    node = null;
+    programState = null;
+    constraintManager = null;
+  }
+
+  private void executeInstruction(BytecodeCFGBuilder.Instruction instruction) {
+    System.out.println("instruc "+instruction.opcode+"  "+ Printer.OPCODES[instruction.opcode]);
+    // TODO execute check prestatement
+    switch (instruction.opcode) {
+      case Opcodes.ACONST_NULL:
+        programState = programState.stackValue(SymbolicValue.NULL_LITERAL);
+        break;
+      case Opcodes.ALOAD:
+      case Opcodes.DLOAD:
+      case Opcodes.FLOAD:
+      case Opcodes.ILOAD:
+        break;
+      case Opcodes.LLOAD:
+      case Opcodes.AALOAD:
+      case Opcodes.BALOAD:
+      case Opcodes.CALOAD:
+      case Opcodes.DALOAD:
+      case Opcodes.FALOAD:
+      case Opcodes.IALOAD:
+      case Opcodes.LALOAD:
+      case Opcodes.SALOAD:
+        break;
+      default:
+        // do nothing
+    }
+    // TODO execute check poststatement
+    enqueue(programPosition.next(), programState);
+  }
+
+  private void handleBlockExit(ProgramPoint programPosition) {
+    programPosition.block.successors().forEach(b -> enqueue(new ProgramPoint(b), programState));
+  }
+
+  private void executeCheckEndOfExecution() {
+    // TODO callback to checks at end of execution
+  }
+
+  private Iterable<ProgramState> startingStates(Symbol.MethodSymbol symbol, ProgramState currentState) {
+    // TODO : deal with parameter annotations, equals methods etc.
+    return Collections.singletonList(currentState);
+  }
+
+  private void setNode(ExplodedGraph.Node node) {
+    this.node = node;
+    programPosition = this.node.programPoint;
+    programState = this.node.programState;
+  }
+
+  private void enqueue(ProgramPoint pp, ProgramState programState) {
+    int nbOfExecution = programState.numberOfTimeVisited(pp);
+    if (nbOfExecution > MAX_EXEC_PROGRAM_POINT) {
+        return;
+      }
+    ProgramState ps = programState.visitedPoint(pp, nbOfExecution + 1);
+    ExplodedGraph.Node cachedNode = explodedGraph.node(pp, ps);
+    cachedNode.addParent(node, null);
+    if (cachedNode.isNew) {
+      workList.addFirst(cachedNode);
+    }
+  }
+
+  private void handleEndOfExecutionPath() {
+    ExplodedGraph.Node savedNode = node;
+    endOfExecutionPath.forEach(n -> {
+      setNode(n);
+//      if (!programState.exitingOnRuntimeException()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)